generic_handler.go

```diff
@@ -4,49 +4,56 @@ import (
 	"github.com/atlassian/ctrl/logz"
 	"go.uber.org/zap"
 	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/tools/cache"
 )
 
 // This handler assumes that the Logger already has the obj_gk/ctrl_gk field set.
 type GenericHandler struct {
-	Logger       *zap.Logger
-	WorkQueue    WorkQueueProducer
-	ZapNameField ZapNameField
+	Logger    *zap.Logger
+	WorkQueue WorkQueueProducer
+
+	Gvk schema.GroupVersionKind
 }
 
 func (g *GenericHandler) OnAdd(obj interface{}) {
-	g.add(obj.(meta_v1.Object), "added")
+	logger := g.Logger.With(logz.Operation("added"))
```

Comment:
Can use the constant here?

btw, why not give all the operations a type (`type Operation string`) and make `logz.Operation()` accept it? Probably move it somewhere... see if it makes sense.

Reply:
I'll need to restructure this PR a bit to do that - there's going to be an import cycle unless I put `Operation` into the logz package.